### PR TITLE
Restart network service instead of ifdown to avoid losing connectivity

### PR DESCRIPTION
**Pull request by mcornea**
To avoid losing network connectivity when bridging the physical interface
to the baremetal bridge we can use systemctl restart network. This is
useful for persisting the SSH connection when automating the deployment.

Also when the baremetal bridge is not managed by libvirt the provsioning host
uses the external DNS server for name resolution. This change sets the
condition so NetworkManager dnsmasq gets used only when the baremetal bridge
is managed by libvirt.

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/447/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/448/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/449/

**Comment by mcornea** [comment]
@derekhiggins Hi! Could you please check this change when you get a moment? Thanks!

File: 02_configure_host.sh [hunk 1]
```diff
@@ -73,14 +73,14 @@ sudo ifup baremetal
...
     echo -e "DEVICE=$INT_IF\nTYPE=Ethernet\nONBOOT=yes\nNM_CONTROLLED=no\nBRIDGE=baremetal" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-$INT_IF
-    sudo ifdown $INT_IF || true
-    sudo ifup $INT_IF
     if sudo nmap --script broadcast-dhcp-discover -e $INT_IF | grep "IP Offered" ; then
         echo -e "\nBOOTPROTO=dhcp\n" | sudo tee -a /etc/sysconfig/network-scripts/ifcfg-baremetal
-        sudo ifdown baremetal || true
-        sudo ifup baremetal
+        sudo systemctl restart network
+    else
+        sudo systemctl restart network
```
Comment (derekhiggins) [inline 02_configure_host.sh]: nit: We have this line at the end of both the "if" and the "else", we could have it once, after the "if" statement

**Pull request merged**